search-app/test/cmr/search/test/data/metadata_retrieval/revision_format_map.clj

```diff
@@ -89,16 +90,32 @@
             #{:echo10 {:format :umm-json
                        :version "1.3"}})))))
 
+(defn- drop-in
+ "Drop a value from a nested structure"
+ [data path keys]
+ (update-in data path (fn [nested] (apply dissoc nested keys )) ))
+
 (deftest ^:kaocha/pending add-additional-format-test
   (testing "Decompressed"
     (let [rfm (r/concept->revision-format-map nil tm/dif10-concept #{:native})]
       (is (= (assoc rfm :echo10 (:metadata tm/echo10-concept))
              (r/add-additional-format nil :echo10 rfm)))))
   (testing "Compressed"
     (let [rfm (r/compress
-                (r/concept->revision-format-map nil tm/dif10-concept #{:native}))]
-      (is (= (assoc rfm :echo10 (util/string->lz4-bytes (:metadata tm/echo10-concept)))
-             (r/add-additional-format nil :echo10 rfm))))))
+                (r/concept->revision-format-map nil tm/dif10-concept #{:native}))
+          expected (assoc rfm :echo10 (util/string->lz4-bytes (:metadata tm/echo10-concept)))
+          actual (r/add-additional-format nil :echo10 rfm)]
+
+      ; A test like (is(= expected actual)) does not work for everyone locally
+      ; as there seams to be some problem with testing byte arrays.
+      ; To get around this, a new test was come up with which drops the
+      ; compressed segment (byte array) which fails and tests that data sepratly
+      ; by doing a comparison on the md5 values
+
+      (is (= (drop-in expected [:echo10] [:compressed])
+             (drop-in actual [:echo10] [:compressed])))
+      (is (= (digest/md5 (String. (get-in expected [:echo10 :compressed]) "UTF-8"))
+             (digest/md5 (String. (get-in actual [:echo10 :compressed] )"UTF-8")))))))
 
```

Comment:
If you are converting the get-in result to a string - you should use the (str (get-in ...)) function instead - it is the proper clojure way.  Calling the java classes is considered not good practice unless you don't have another choice.

Reply:
I tried to use str but I think it was encoding the hash value of the string object and not the content of the string. md5 is wanting an actual java string, str was confusing it.

Reply:
OK